### PR TITLE
Add support for map and parametric types

### PR DESCRIPTION
**Pull request by zbynek**
This new parameter type could be consumed by the steps docs generator or snippetizer.

### CODE REVIEW

**Comment by zbynek** [comment]
@car-roll could you please check this PR?

**Comment by car-roll** [comment]
Hi @zbynek I this PR is trying to do the same as https://github.com/jenkinsci/structs-plugin/pull/52
There were a couple of important points raised in that discussion, it would probably be worthwhile to take a look at the various discussions that happened over there. In particular, the ones here:
https://github.com/jenkinsci/structs-plugin/pull/52#pullrequestreview-33881178
https://github.com/jenkinsci/structs-plugin/pull/52#issuecomment-620024158

Long story short, `Map` is not a legal binding type and adding support for map and other parameterized types has implications that go beyond this plugin
